File: Euclid/Messages/Outgoing/ACTIVEOBJECT_ADD.cs
using Euclid.Game;
using Euclid.Network.Streams.Util;
using System.Collections.Generic;

namespace Euclid.Messages.Outgoing
{
    public class ACTIVEOBJECT_ADD : IMessageComposer
    {
        private Item item;

        public ACTIVEOBJECT_ADD(Item item)
        {
            this.item = item;
        }

        public override void Write()
        {
            this.item.Serialise(Data);
        }
    }
}

File: euclid/src/main/java/euclid/messages/outgoing/ACTIVEOBJECT_ADD.java
package euclid.messages.outgoing;

import euclid.game.item.Item;
import euclid.messages.IMessageComposer;
import euclid.network.streams.util.ArgumentEntry;
import euclid.network.streams.util.DelimeterEntry;

public class ACTIVEOBJECT_ADD extends IMessageComposer {

    private final Item item;

    public ACTIVEOBJECT_ADD(Item item) {
        this.item = item;
    }

    @Override
    public void write() {
        item.serialise(getData());
    }
}
